_posts/2018-08-19-openfaas-on-fargate.md

```diff
@@ -0,0 +1,281 @@
+---
+title: Faas on Fargate&#58; run OpenFaaS on AWS Fargate
+description: Edward Wilde introduces running OpenFaaS on AWS Fargate - a simple, low cost way to deploy and run OpenFaaS functions without managing servers
+date: 2019-02-04
+image: /images/openfaas-on-fargate/astronomy-atmosphere-earth-220201.jpg
+categories:
+  - Fargate
+  - Automation
+author_staff_member: ed
+dark_background: true
+---
+
+How to run OpenFaaS on AWS Fargate with economical, auto-scaling containers. A low friction AWS
+native deployment using VPCs, ECS, security groups, ALBs, AWS Secrets Manager and AWS Route 53 for
+DNS, load-balancing and service discovery
+
+_Experience level:_
+
+_beginner OpenFaaS / intermediate AWS Fargate_
+
+This article will explain why I've chosen to run OpenFaaS on AWS Fargate. We then move on to explore
+how to deploy a fully working environment and run our first function.
+
+One of the great things about OpenFaaS is it's portability. You can run OpenFaaS out of the box using 
+officially supported platforms like are Kubernetes and Docker Swarm . The [community](https://github.com/openfaas/faas/blob/master/community.md) 
+has added many other platforms, that allow you to run OpenFaaS in more places. A good example of this is deploying OpenFaaS
+on Hashicorp's [nomad](https://www.hashicorp.com/blog/functions-as-a-service-with-nomad) application scheduler.
+
+### Why use AWS Fargate?
+Your reasons for choosing AWS Fargate are likely to vary depending on your
+experience, knowledge and goals. Perhaps you work in a team that has little or no knowledge of application schedulers like
+Kubernetes or Nomad. You may also work in a company that hasn't adopted Kubernetes and doesn't plan to in the immediate 
+future. So if you fit that in that camp, how do you leverage the advantage of serverless computing? A good choice would
+be to look at a managed serverless platform like [AWS Lambda](https://aws.amazon.com/lambda/). In this article we examine 
+another option, running OpenFaaS on Fargate, which I believe gives you some of the benefits of a managed service like 
+lambda with the flexibility of an open-source serverless platform like OpenFaaS.
+
+## Architecture
+OpenFaaS functions are built as docker images and deployed as docker containers. This approach will be very familiar to
+developers using microservices. The platform is composed of a `gateway`, `provider` and `monitoring` components. The 
```

Comment:
"are packaged as Docker images" "Docker containers"

Reply:
> This approach will be very familiar to developers using microservices.

> If you are already using microservices, you may find this approach very similar.

Let's suggest rather than tell.